src/resmom/mom_main.c

```diff
@@ -6437,6 +6445,919 @@ kill_job(job *pjob, int sig)
+			if (val.at_type == ATR_TYPE_STR && pjob->ji_numnodes == 1) {
+				/* check if string value is a valid json string,
+				 * if it is then set the resource string within
+				 * single quotes.
+				 */
+
+				sval = val.at_val.at_str;
+				if ((py_jvalue = json_loads(sval, emsg, HOOK_BUF_SIZE - 1)) != NULL) {
+					dumps = json_dumps(py_jvalue, emsg, HOOK_BUF_SIZE - 1);
+					if (dumps == NULL)
+						Py_CLEAR(py_jvalue);
+					else {
+						rd->rs_decode(&tmpatr, ATTR_used, rd->rs_name, dumps);
+						val = tmpatr;
+						Py_CLEAR(py_jvalue);
+						free(dumps);
+						dumps = NULL;
+					}
+				}
+			}
+
+			if (rd->rs_encode(&val, phead, ad->at_name, rd->rs_name, ATR_ENCODE_CLIENT, NULL) < 0)
+				goto encode_used_exit;
+
+			if (include_resc_used_update) {
+				if (rd->rs_encode(&val3, phead, ad3->at_name, rd->rs_name, ATR_ENCODE_CLIENT, NULL) < 0)
+					goto encode_used_exit;
+			}
+		}
+
+encode_used_exit:
+		if ((tmpatr.at_flags & ATR_VFLAG_SET) != 0 && tmpatr.at_type == ATR_TYPE_STR)
+			rd->rs_free(&tmpatr);
+		if ((tmpatr3.at_flags & ATR_VFLAG_SET) != 0 && tmpatr3.at_type == ATR_TYPE_STR)
+			rd->rs_free(&tmpatr3);
+	}
+}
+
+static int
+is_in_old_ruu(ruu *old, char *name)
+{
+	if (old == NULL)
+		return 0;
+	if (find_svrattrl_list_entry(&old->ru_attr, name, NULL) != NULL)
+		return 1;
+	return 0;
+}
+
+ruu *
+get_job_update(job *pjob)
+{
+	/*
+	 * the following is a list of attributes to be returned to the server
+	 * for a newly executing job. They are returned only if they have
+	 * been modified by Mom.  Note that JOB_ATR_session_id and JOB_ATR_resc_used
+	 * are always returned;
+	 */
+	static enum job_atr mom_rtn_list[] = {
+		JOB_ATR_errpath,
+		JOB_ATR_outpath,
+		JOB_ATR_altid,
+		JOB_ATR_acct_id,
+		JOB_ATR_jobdir,
+		JOB_ATR_exectime,
+		JOB_ATR_hold,
+		JOB_ATR_variables,
+		JOB_ATR_runcount,
+		JOB_ATR_exec_vnode,
+		JOB_ATR_SchedSelect,
+		JOB_ATR_LAST
+	};
+	ruu *prused;
+	int i;
+	int nth;
+	attribute *at;
+	attribute_def *ad;
+
+	prused = (ruu *) calloc(1, sizeof(ruu));
+	if (prused == NULL) {
+		log_joberr(errno, __func__, "Out of memory while encoding stat update", pjob->ji_qs.ji_jobid);
+		return NULL;
+	}
+	CLEAR_LINK(prused->ru_pending);
+	CLEAR_HEAD(prused->ru_attr);
+	prused->ru_created_at = time(0);
+	prused->ru_pjobid = strdup(pjob->ji_qs.ji_jobid);
+	if (prused->ru_pjobid == NULL) {
+		FREE_RUU(prused);
+		log_joberr(errno, __func__, "Out of memory while encoding jobid in stat update", pjob->ji_qs.ji_jobid);
+		return NULL;
+	}
+
+	resc_access_perm = ATR_DFLAG_MGRD;
+
+	if (pjob->ji_wattr[JOB_ATR_run_version].at_flags & ATR_VFLAG_SET) {
+		prused->ru_hop = pjob->ji_wattr[JOB_ATR_run_version].at_val.at_long;
+	} else {
+		prused->ru_hop = pjob->ji_wattr[JOB_ATR_runcount].at_val.at_long;
+	}
+#ifdef WIN32
+	if (pjob->ji_wattr[JOB_ATR_Comment].at_flags & ATR_VFLAG_SET) {
+		prused->ru_comment = strdup(pjob->ji_wattr[JOB_ATR_Comment].at_val.at_str);
+		if (prused->ru_comment == NULL) {
+			FREE_RUU(prused);
```

Comment:
@prakashcv13 assuming you are return NULL if we fails to encode comment in ruu, then we are already doing that. If you are asking not to return NULL and continue without comment in ruu, then I am not sure about that, need to check on windows as sending comment is only for windows not linux.

Reply:
i was just thinking that setting of comment is not something that should stop us from continuing. If it makes more sense to not continue, please feel free to ignore my comment.

Reply:
Thanks for confirming @prakashcv13, I will double check on windows that without comment is not giving any problem, and will change accordingly

Reply:
So I checked this, and its turns out that comment in updates is not used at all in stat_update but only used while job_obit and that only for one case, which is if job fails due to bad password and on server side job goes in hold state, so I think we can send update if strdup() fails as it will get included in next updates (like other attrs)...